src/core/qgsattributetableconfig.h

```diff
@@ -190,6 +190,11 @@ class CORE_EXPORT QgsAttributeTableConfig
      */
     void setSortOrder( Qt::SortOrder sortOrder );
 
+    /**
+     * Compare this configuration's columns name and type to other.
```

Comment:
"\a other". 

I think this needs expanding a bit - it should explicitly state that other settings (such as column width) are not considered. 

The name should be hasSameColumns too, to match the Qt convention.

Reply:
Oh, and I'd say in the docs that column order is tested too.

Reply:
(^^ better add a unit test to make sure this method considers column order too)